src/boot/efi/cpio.c

```diff
@@ -336,9 +337,23 @@ EFI_STATUS pack_cpio(
                 return EFI_SUCCESS;
         }
 
-        root = LibOpenRoot(loaded_image->DeviceHandle);
-        if (!root)
-                return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
+        err = BS->HandleProtocol(loaded_image->DeviceHandle,
+                                 &FileSystemProtocol, (void*)&volume);
+        /* EFI only supports vfat, so the error will be "unsupported" if the
+         * image was loaded from a non-vfat partition, which might happen if,
+         * say, UEFI loaded grub from an ESP, but then grub loaded the image
+         * from a different, non-vfat partition. Do not stall in that case.
```

Comment:
`Do not warn in that case`. The stalling is just a "side-effect" :D

I also think the mention of only vfat being supported should go. I'm pretty sure ASUS will happily boot from NTFS. The only important thing is whether the image comes from something that doesn't support the protocol that we need to read the files.